rclcpp_lifecycle/src/lifecycle_node.cpp

```diff
@@ -152,8 +152,19 @@ LifecycleNode::LifecycleNode(
 
 LifecycleNode::~LifecycleNode()
 {
+  auto current_state = LifecycleNode::get_current_state().id();
+  if (current_state != lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED) {
+    // This might be leaving sensors and devices without shutting down unintentionally.
+    // It is user's responsibility to call shutdown to avoid leaving them unknow states.
+    RCLCPP_WARN(
+      rclcpp::get_logger("rclcpp_lifecycle"),
```

Comment:
minor: it should be safe to use the node-specific logger here (`this->get_logger()`), since the logging interface hasn't been shut down yet